src/main/java/pl/touk/sputnik/processor/shellcheck/ShellcheckExecutor.java

```diff
@@ -0,0 +1,47 @@
+package pl.touk.sputnik.processor.shellcheck;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import pl.touk.sputnik.exec.ExternalProcess;
+
+import java.util.List;
+
+@Slf4j
+class ShellcheckExecutor {
+    private static final String SHELLCHECK_EXECUTABLE = "shellcheck";
+    private static final String SHELLCHECK_OUTPUT_FORMAT = "-fjson";
+    private static final String SHELLCHECK_EXCLUDE_RULES = "-e";
+
+    private final List<String> excludeRules;
+
+    ShellcheckExecutor(List<String> excludeRules) {
+        this.excludeRules = excludeRules;
+    }
+
+    String runOnFile(String filePath) {
+        log.info("Review on file: " + filePath);
+        return new ExternalProcess().executeCommand(buildParams(filePath));
```

Comment:
What if shellcheck is not installed?

Reply:
Then ExternalPorcess throws **ExternalProcessException** which is not handled specially like in the case of other ProcessorRunningExternalProcess processors